### ISSUE TITLE
pip3 install sexpdata caught error

### ISSUE THREAD

**Issue by dreamkidd** [jd-boyd/sexpdata#51]
hello , my python version is  **3.11.5** and pip version is pip 23.3.1 

when i install via `pip3 install sexpdata` , i got error message blow

```
❯ pip3 install sexpdata
Collecting sexpdata
  Using cached sexpdata-1.0.1.tar.gz (8.6 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [75 lines of output]
      /private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py:75: _MissingDynamic: `license` defined outside of `pyproject.toml` is ignored.
      !!

              ********************************************************************************
              The following seems to be defined outside of `pyproject.toml`:

              `license = 'BSD License'`

              According to the spec (see the link below), however, setuptools CANNOT
              consider this value unless `license` is listed as `dynamic`.

              https://packaging.python.org/en/latest/specifications/declaring-project-metadata/

              To prevent this problem, you can list `license` under `dynamic` or alternatively
              remove the `[project]` table from your file and rely entirely on other means of
              configuration.
              ********************************************************************************

      !!
        _handle_missing_dynamic(dist, project_table)
      /private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py:75: _MissingDynamic: `keywords` defined outside of `pyproject.toml` is ignored.
      !!

              ********************************************************************************
              The following seems to be defined outside of `pyproject.toml`:

              `keywords = ['s-expression', 'lisp', 'parser']`

              According to the spec (see the link below), however, setuptools CANNOT
              consider this value unless `keywords` is listed as `dynamic`.

              https://packaging.python.org/en/latest/specifications/declaring-project-metadata/

              To prevent this problem, you can list `keywords` under `dynamic` or alternatively
              remove the `[project]` table from your file and rely entirely on other means of
              configuration.
              ********************************************************************************

      !!
        _handle_missing_dynamic(dist, project_table)
      Traceback (most recent call last):
        File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 8, in <module>
        File "/private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 159, in setup
          dist.parse_config_files()
        File "/private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 627, in parse_config_files
          pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
        File "/private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/config/pyprojecttoml.py", line 67, in apply_configuration
          return _apply(dist, config, filepath)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py", line 56, in apply
          _apply_project_table(dist, config, root_dir)
        File "/private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py", line 82, in _apply_project_table
          corresp(dist, value, root_dir)
        File "/private/var/folders/fj/31b93k8j72b0723dj4cj4vk00000gn/T/pip-build-env-rawt4j2c/overlay/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py", line 183, in _license
          _set_config(dist, "license", val["text"])
                                       ~~~^^^^^^^^
      KeyError: 'text'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

```

but if i use `pip3 install sexpdata==1.0.0`  , i will get no error and installed successful

**Comment by spogelse** [comment]
This was actually fixed in #45.
@jd-boyd Could you please push this to pypi.org? Sexpdata is a dependency of epc, which is used for Python completion in Emacs. Obviously its installation is currently broken because it tries to pull in sexpdata 1.0.1 from Pypi.

**Comment by yanboyang713** [comment]
I had the same issue. Install sexpdata to arch linux.
```
yanboyang713@Meta-Scientific-Linux ~ % paru -S python-sexpdata
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...
:: orphans: python-sexpdata

Aur (1) python-sexpdata-1.0.1-1

:: Proceed to review? [Y/n]: Y

:: Downloading PKGBUILDs...
 PKGBUILDs up to date
 nothing new to review
fetching devel info...
==> Making package: python-sexpdata 1.0.1-1 (Fri 29 Dec 2023 04:21:11 PM EST)
==> Retrieving sources...
  -> Found sexpdata-1.0.1.tar.gz
==> Validating source files with sha256sums...
    sexpdata-1.0.1.tar.gz ... Passed
==> Making package: python-sexpdata 1.0.1-1 (Fri 29 Dec 2023 04:21:12 PM EST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found sexpdata-1.0.1.tar.gz
==> Validating source files with sha256sums...
    sexpdata-1.0.1.tar.gz ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting sexpdata-1.0.1.tar.gz with bsdtar
==> Sources are ready.
python-sexpdata-1.0.1-1: parsing pkg list...
==> Making package: python-sexpdata 1.0.1-1 (Fri 29 Dec 2023 04:21:13 PM EST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Removing existing $pkgdir/ directory...
==> Starting build()...
* Getting build dependencies for wheel...
validate_pyproject.api.load_builtin_plugin defines `tool.distutils` schema
validate_pyproject.api.load_builtin_plugin defines `tool.setuptools` schema
/usr/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py:75: _MissingDynamic: `license` defined outside of `pyproject.toml` is ignored.
!!

        ********************************************************************************
        The following seems to be defined outside of `pyproject.toml`:

        `license = 'BSD License'`

        According to the spec (see the link below), however, setuptools CANNOT
        consider this value unless `license` is listed as `dynamic`.

        https://packaging.python.org/en/latest/specifications/declaring-project-metadata/

        To prevent this problem, you can list `license` under `dynamic` or alternatively
        remove the `[project]` table from your file and rely entirely on other means of
        configuration.
        ********************************************************************************

!!
  _handle_missing_dynamic(dist, project_table)
/usr/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py:75: _MissingDynamic: `keywords` defined outside of `pyproject.toml` is ignored.
!!

        ********************************************************************************
        The following seems to be defined outside of `pyproject.toml`:

        `keywords = ['s-expression', 'lisp', 'parser']`

        According to the spec (see the link below), however, setuptools CANNOT
        consider this value unless `keywords` is listed as `dynamic`.

        https://packaging.python.org/en/latest/specifications/declaring-project-metadata/

        To prevent this problem, you can list `keywords` under `dynamic` or alternatively
        remove the `[project]` table from your file and rely entirely on other means of
        configuration.
        ********************************************************************************

!!
  _handle_missing_dynamic(dist, project_table)
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
    main()
  File "/usr/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 335, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
    return hook(config_settings)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
    return self._get_build_requires(config_settings, requirements=['wheel'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
    self.run_setup()
  File "/usr/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
    exec(code, locals())
  File "<string>", line 8, in <module>
  File "/usr/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 159, in setup
    dist.parse_config_files()
  File "/usr/lib/python3.11/site-packages/setuptools/dist.py", line 627, in parse_config_files
    pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
  File "/usr/lib/python3.11/site-packages/setuptools/config/pyprojecttoml.py", line 73, in apply_configuration
    return _apply(dist, config, filepath)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py", line 56, in apply
    _apply_project_table(dist, config, root_dir)
  File "/usr/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py", line 82, in _apply_project_table
    corresp(dist, value, root_dir)
  File "/usr/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py", line 183, in _license
    _set_config(dist, "license", val["text"])
                                 ~~~^^^^^^^^
KeyError: 'text'

ERROR Backend subprocess exited when trying to invoke get_requires_for_build_wheel
==> ERROR: A failure occurred in build().
    Aborting...
error: failed to build 'python-sexpdata-1.0.1-1':
error: packages failed to build: python-sexpdata-1.0.1-1
```

**Comment by HappyCthulhu** [comment]
Manjaro, same error. Do u have some updates about with topic?

```
[N] valera@t-t14 ~> pipx install python-sexpdata
Fatal error from pip prevented installation. Full pip output in file:
    /home/valera/.local/pipx/logs/cmd_2024-01-01_16.48.01_pip_errors.log

Some possibly relevant errors from pip install:
    ERROR: Could not find a version that satisfies the requirement python-sexpdata (from versions: none)
    ERROR: No matching distribution found for python-sexpdata

Error installing python-sexpdata.
[N] valera@t-t14 ~ [1]> pipx install sexpdata
Fatal error from pip prevented installation. Full pip output in file:
    /home/valera/.local/pipx/logs/cmd_2024-01-01_16.48.11_pip_errors.log

pip seemed to fail to build package:
    sexpdata

Some possibly relevant errors from pip install:
    error: subprocess-exited-with-error
    KeyError: 'text'

Error installing sexpdata.
[I] valera@t-t14 ~ [1]>
```

**Comment by niontrix** [comment]
As a workaround you can install sexpdata using version 1.0.0, before trying to install epc. That worked for me:
`pip install "sexpdata==1.0.0"`

**Comment by jd-boyd** [comment]
I think this should be fixed with the 1.0.2 release.
